### PR TITLE
[WIP] Convert project to gradle build system

### PR DESCRIPTION
**Pull request by WeaponMan**
Almost everything is working, but publishing to maven repository and pom.xml customization is little riddle for me.

I will update this soon, if you think this is good idea.

### CODE REVIEW

**Comment by sockeqwe** [comment]
Thanks a lot! I will take care of maven publishing, i did it 100 times before, I have it within a minute ...

**Comment by WeaponMan** [comment]
Nice!
Maybe findbugs plugin will need some care, because I am not completely sure if I configured all modules exactly same as maven ones.

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,212 @@
+buildscript {
```
Comment (FabianTerhorst) [inline build.gradle]: I would also prefer one root build.gradle file and one file for each sub project. It the more common and cleaner solution.

Comment (WeaponMan) [inline build.gradle]: I did it because all configuration is in one place (faster to edit).

**Comment by WeaponMan** [comment]
Splitted version is live. Better?

**Comment by WeaponMan** [comment]
I found that apt plugin only work in commandline build.
In IDEA build the classes are not generated at the build time.

**Comment by sockeqwe** [comment]
apt + gradle was historically always cumbersome to work with (one of the
reasons why I used maven).

I hope that it works with this
https://github.com/tbroyer/gradle-apt-plugin

in combination with

https://docs.gradle.org/3.3/userguide/idea_plugin.html

I will try it at the end of June ...

Ondřej Kotora <notifications@github.com> schrieb am Di., 20. Juni 2017,
09:19:

**Comment by WeaponMan** [comment]
I tried that yesterday on my java sample with no success using IntelliJ IDEA 2017.1.3 UE.
With idea plugin and `./gradlew idea` same result.
Maybe this requires more configuration.

**Comment by iNoles** [comment]
Did you apply the idea plugin too?

**Comment by WeaponMan** [comment]
Yes something like that:
```groovy
plugins {
    id 'java'
    id 'idea'
    id "net.ltgt.apt" version "0.10"
    id 'com.github.johnrengelman.shadow' version '2.0.0'
}
```

**Comment by iNoles** [comment]
Well, Kotlin has own apt too which called "kapt."

https://kotlinlang.org/docs/reference/using-gradle.html

**Comment by WeaponMan** [comment]
I removed all idea project settings and re-imported the project to idea again closely following this:
https://github.com/tbroyer/gradle-apt-plugin

But skipped part with idea gradle task.
Its finally working as it should.

Kapt is maybe worse than that.
I cloned `JetBrains/kotlin-examples` repo and tried dagger sample in idea.
Classes are generated when i run `gradlew assemble` but not when I click build in IDEA.

**Comment by sockeqwe** [comment]
I think we can close it for now. Maven works good enough at the moment and it doesn't add any benefits at the moment to move to gradle. 

What do you think?

**Comment by WeaponMan** [comment]
Its only nicer syntax and familiarity with it, but sure I will close this
for "now".

**Comment by sockeqwe** [comment]
Anyways, thanks for your hard work. We might convert it in the future.

**Pull request closed without merging**